cypress/support/MonitorDocumentChanges.js

```diff
@@ -0,0 +1,219 @@
+            {
+                //helpers.ConLog(`cy.should())`, `Waiting for Stable DOM - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago - subject: ${subject} - chainers: ${chainers} - value: ${value}`)
+                return originalFn(subject, chainers, MillisecondsSinceLastChange())
+            }
+            return originalFn(subject, chainers, value)
+        })
+
+        Cypress.Commands.add('Get', (selector) => 
+        {   
+            helpers.ConLog(`cy.Get()`, `Start - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago - Selector: \n${selector}`)
+            cy.wrap(700, {timeout: 60000}).should('lte', 'MillisecondsSinceLastChange').then(() => {
+            helpers.ConLog(`cy.Get()`, `DOM Is Stable`)
+            cy.get(selector)
+        })})
+
+        // Special case version that allows a 1 minute time out.
+        Cypress.Commands.add('Contains', (selector, content, options) => 
+        {   
+            helpers.ConLog(`cy.Contains()`, `Start - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago - Selector -- Content: \n${selector} -- ${content}`)
+            cy.wrap(700, {timeout: 60000}).should('lte', 'MillisecondsSinceLastChange').then(() => {
+            helpers.ConLog(`cy.Contains()`, `DOM Is Stable`)
+            cy.contains(selector, content, options)
+        })})
+
+        Cypress.Commands.add('Click', { prevSubject: true, element: true}, (subject) => 
+        {
+            helpers.ConLog(`cy.Click()`, `Start - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago`)
+            lastChangeTime = new Date().getTime()
+            cy.wrap(subject).click()
+            .then(()=> {helpers.ConLog(`cy.Click()`, `done - Last DOM change was ${MillisecondsSinceLastChange()} milliseconds ago`)
+            lastChangeTime = new Date().getTime()})
+        })
+
+        Cypress.Commands.add('DumpHtmlOnDomChange', (boolValue) => {dumpHtml = boolValue})
+
+        Cypress.Commands.add('VerifyMonitorFinds', () => { expect(canRefreshTrainingStatusTime).to.equal(0) })
+
+        Cypress.on('window:before:load', () => 
+        { 
+            helpers.ConLog(`window:before:load`, `MillisecondsSinceLastChange: ${MillisecondsSinceLastChange()}`) 
+            lastChangeTime = new Date().getTime()
+        })
+
+        Cypress.on('window:load', () => 
+        { 
+            helpers.ConLog(`window:load(complete)`, `MillisecondsSinceLastChange: ${MillisecondsSinceLastChange()}`) 
+            lastChangeTime = new Date().getTime()
+        })
+
+        Cypress.on('url:changed', (newUrl) => 
+        {
+            helpers.ConLog(`url:changed`, `New URL ${newUrl} - MillisecondsSinceLastChange: ${MillisecondsSinceLastChange()}`) 
+            lastChangeTime = new Date().getTime()
+            
+            // TODO: Remove this UrlNeedsSpinner() function in time if it proves that we no longer need it.
```

Comment:
Looks like it was removed?   Can this go or still testing?

Reply:
I am still testing this. This failure this was meant to fix happened occassionally. I've not done enough testing to be confident my other fix is fully working yet.